packages/valory/skills/abstract_round_abci/behaviour_utils.py

```diff
@@ -1699,6 +1707,86 @@ def __parse_rpc_error(error: str) -> RPCResponseStatus:
             return RPCResponseStatus.ALREADY_KNOWN
         return RPCResponseStatus.UNCLASSIFIED_ERROR
 
+    def _acn_request_from_pending(
+        self, performative: TendermintMessage.Performative
+    ) -> Generator:
+        """Perform an ACN request to each one of the agents which have not sent a response yet."""
+        not_responded_yet = {
+            address
+            for address, deliverable in cast(
+                SharedState, self.context.state
+            ).address_to_acn_deliverable.items()
+            if deliverable is None
+        }
+
+        if len(not_responded_yet) == 0:
+            return
+
+        self.context.logger.debug(f"Need ACN response from {not_responded_yet}.")
+        for address in not_responded_yet:
+            self.context.logger.debug(f"Sending ACN request to {address}.")
+            dialogues = cast(TendermintDialogues, self.context.tendermint_dialogues)
+            message, _ = dialogues.create(
+                counterparty=address, performative=performative
+            )
+            message = cast(TendermintMessage, message)
+            context = EnvelopeContext(connection_id=P2P_LIBP2P_CLIENT_PUBLIC_ID)
+            self.context.outbox.put_message(message=message, context=context)
+
+        # we wait for the `address_to_acn_deliverable` to be populated with the responses (done by the tm handler)
+        yield from self.sleep(self.params.sleep_time)
+
+    def _perform_acn_request(
+        self, performative: TendermintMessage.Performative
+    ) -> Generator[None, None, Any]:
+        """Perform an ACN request.
+
+        Waits `sleep_time` to receive a common response from the majority of the agents.
+        Retries `max_attempts` times only for the agents which have not responded yet.
+
+        :param performative: the ACN request performative.
+        :return: the result that the majority of the agents sent. If majority cannot be reached, returns `None`.
+        """
+        # reset the ACN deliverables at the beginning of a new request
+        addresses = self.synchronized_data.all_participants
+        cast(
+            SharedState, self.context.state
+        ).address_to_acn_deliverable = dict.fromkeys(addresses)
+
+        for i in range(self.params.max_attempts):
+            self.context.logger.debug(
+                f"ACN attempt {i + 1}/{self.params.max_attempts}."
+            )
+            yield from self._acn_request_from_pending(performative)
+
+            result = cast(SharedState, self.context.state).get_acn_result()
```

Comment:
Should we set this to None after getting it once?

Reply:
Do you mean the `address_to_acn_deliverable`? This is reset every time we begin a new ACN request at the start of this method.